### PR TITLE
feature: events (integrated)

### LINKED ISSUE #30: Events (integrated)
Events will be shown on the landing page as on the Figma design.

### PR DESCRIPTION
**Pull request by PradeepKhatri**
Fixes #30

### CODE REVIEW

File: app/events.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            <div className="text-center text-red-700 text-3xl font-normal font-serif leading-10">
+              Student
+            </div>
+          </div>
+          <div className="self-stretch grow shrink basis-0 px-9 py-6 bg-white rounded-xl border border-red-700 justify-center items-center gap-2 inline-flex">
+            <div className="text-center text-red-700 text-3xl font-normal font-serif leading-10">
+              Faculty
+            </div>
+          </div>
+        </div>
```
Comment (Maan2003) [inline app/events.tsx:73]: desperate need of components

**Review by GetPsyched — CHANGES_REQUESTED** [review]
1. Images will be fetched from Figma (later S3), remove them from the Git tree.
2. Rebase on master to get the latest stuff.
3. Use semantic HTML tags instead of `div` and `span`. Those should be your last resort.
4. Make an object for all the data, they will most likely be fetched dynamically from the database.

File: styles/globals.css
Comment (GetPsyched) [inline styles/globals.css:1]: Please don't remove existing styling.

File: styles/globals.css [hunk 2]
```diff
@@ -5,36 +5,30 @@
 @tailwind components;
 @tailwind utilities;
 
+@import url('https://fonts.googleapis.com/css2?family=DM+Serif+Display:wght@400&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=DM+Sans:wght@400;700&display=swap');
+
+@import url('https://fonts.googleapis.com/css2?family=DM+Serif+Display&display=swap');
+
```
Comment (GetPsyched) [inline styles/globals.css:12]: We already have the fonts installed.
```suggestion
```

File: tailwind.config.ts
Comment (GetPsyched) [inline tailwind.config.ts:1]: Another case of removing existing styling

File: tsconfig.json [hunk 3]
```diff
@@ -26,6 +26,12 @@
       "@/*": ["./*"],
     },
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    ".next/types/**/*.ts",
+    "env.mjs",
```
Comment (GetPsyched) [inline tsconfig.json:34]: This had been removed. Don't add it back (happened due to bad resolution of merge conflicts during rebase)
```suggestion
```

File: app/[locale]/page.tsx [hunk 4]
```diff
@@ -8,8 +9,7 @@ export default function Home({
 }) {
   return (
     <main className="w-full">
-      <Notifications />
-      <DirectorsCorner locale={locale} />
+      <Events />
```
Comment (GetPsyched) [inline app/[locale]/page.tsx:12]: Don't remove the other components.
```suggestion
      <Notifications />
      <Events /> 	
      <DirectorsCorner locale={locale} />
```

File: components/horsesRunning.tsx
Comment (GetPsyched) [inline components/horsesRunning.tsx:1]: Uh, this already exists, why adding it again?

Comment (GetPsyched) [inline components/horsesRunning.tsx:1]: Uh, this component already exists, why adding it again?

Comment (GetPsyched) [inline components/horsesRunning.tsx:1]: Uh, this component already exists, why adding it again?

File: app/[locale]/events.tsx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    imageWidth: 753,
+    imageHeight: 300,
+    link: '#',
+  },
+};
+
+export default function Events() {
+  return (
+    <article className=" my-10 min-w-full lg:container">
+      {/* -----HEADING------ */}
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:68]: Remove these comments

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/[locale]/events.tsx [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+        <section
+          className=""
+          style={{
+            backgroundImage: "url('https://i.imgur.com/Iwh3b4c.png')",
+            backgroundSize: '100% 100%',
+            paddingBottom: '6pc',
+          }}
+        >
+          <div className=" flex justify-end text-primary-20">
+            <button className="mr-[8rem] text-lg font-bold lg:font-sans">
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:92]: Font won't be defined based on a screen width.

File: app/[locale]/events.tsx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+              VIEW ALL
+            </button>
+          </div>
+
+          <div className=" mt-[31px] flex">
+            {/* category buttons */}
+            <div className="ml-[240px] inline-flex h-[490px] w-[282px] flex-col items-start justify-start gap-6">
+              <button
+                className="rounded-xl border-2 bg-shades-10 px-[50px] py-[34px] font-serif text-2xl  leading-10 text-primary-20  hover:bg-shades-30 hover:text-primary-50"
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:101]: No need to configure fonts

File: app/[locale]/events.tsx [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+                className="rounded-xl border-2 bg-shades-10 px-[56px] py-[34px] font-serif text-2xl leading-10 text-primary-20  hover:bg-shades-30 hover:text-primary-50"
+                style={{
+                  boxShadow: '0px 4px 16px 0px rgba(0, 29, 61, 0.20)',
+                }}
+              >
+                Recents
+              </button>
+
+              <button
+                className="rounded-xl border-2 bg-shades-10 px-[56px] py-[34px] font-serif text-2xl leading-10 text-primary-20 hover:bg-shades-30 hover:text-primary-50"
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:119]: Same here

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Leave configuring fonts for now. We will do it globally.

File: tsconfig.json [hunk 9]
```diff
@@ -26,6 +26,11 @@
       "@/*": ["./*"],
     },
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    ".next/types/**/*.ts",
```
Comment (GetPsyched) [inline tsconfig.json:33]: Remove the comma here so that it goes back to a single line.

File: tsconfig.tsbuildinfo
Comment (GetPsyched) [inline tsconfig.tsbuildinfo:1]: Remove this file from the Git tree. It's not supposed to be pushed

File: tailwind.config.ts
Comment (GetPsyched) [inline tailwind.config.ts:1]: Please don't edit global styles.

File: app/[locale]/events.tsx [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+                className="rounded-xl border border-primary-700 bg-shade-light px-[58px] py-[34px] leading-10 text-primary-700  hover:bg-red-600 hover:text-white"
+                style={{
+                  boxShadow: '0px 4px 16px 0px rgba(0, 29, 61, 0.20)',
+                }}
+              >
+                <h4>Faculty</h4>
+
+              </button>
+            </div> */}
+
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:167]: Remove if redundant

File: app/[locale]/events.tsx [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+          height={9.033 * 16}
+        />
+      </header>
+
+      <section className="h-auto bg-cover">
+        <section
+          style={{
+            backgroundImage: "url('https://i.imgur.com/Iwh3b4c.png')",
+            backgroundSize: '100% 100%',
+            paddingBottom: '6pc',
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:108]: What is `pc`? Is this a typo? Even if not, use standard units.

File: app/[locale]/events.tsx [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+                    height={
+                      Data.Featured.information.interNitTournament.imageHeight
+                    }
+                  />
+
+                  <div className="flex flex-col justify-around gap-2 px-4 py-4">
+                    <h4 className="text-primary-100">
+                      {Data.Featured.information.interNitTournament.title}
+                    </h4>
+                    <p className="">
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:198]:
```suggestion
                    <p>
```

File: app/[locale]/events.tsx [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+                </div>
+
+                <aside className="flex w-[428px] flex-col justify-around space-x-1 text-primary-100">
+                  {Data.Featured.information.advertisements.map((ad, index) => (
+                    <div
+                      key={index}
+                      className="rounded-xl bg-neutral-50 px-4 py-6"
+                    >
+                      <h3>
+                        <a href={ad.link}>{ad.title}</a>
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:211]: Please use `<Link />` from `next/link` instead of `a` tags.

File: app/[locale]/events.tsx [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+                      <Image
+                        src={Data.Featured.information.dangerInKkr.imageUrl}
+                        className="h-[174px] rounded-t-xl"
+                        alt={Data.Featured.information.dangerInKkr.imageAlt}
+                        width={Data.Featured.information.dangerInKkr.imageWidth}
+                        height={
+                          Data.Featured.information.dangerInKkr.imageHeight
+                        }
+                      />
+                    </figure>
```
Comment (GetPsyched) [inline app/[locale]/events.tsx:239]: There's no point wrapping inside a tag if there's only one child.

**Review by GetPsyched — CHANGES_REQUESTED** [review]

**Comment by GetPsyched** [comment]
Addressed by #221

**Pull request closed without merging**